core/modules/file/file.field.inc

```diff
@@ -709,7 +718,7 @@ function file_field_widget_form(&$form, &$form_state, $field, $instance, $langco
     // field. These are added here so that they may be referenced through a
     // hook_form_alter().
     $elements['#file_upload_title'] = t('Add a new file');
```

Comment:
This should account for when multiple files are allowed, and be `t('Add new files')` instead.